frontend/src/pages/CompareV2.tsx

```diff
@@ -14,13 +14,166 @@
+          message: `Error: failed loading ${runIds.length} runs. Click Details for more information.`,
+          mode: 'error',
+        });
+      },
+      onSuccess: () => props.updateBanner({}),
+    },
+  );
+
+  useEffect(() => {
+    const refresh = async () => {
+      if (runlistRef.current) {
+        await runlistRef.current.refresh();
+      }
+      await refetch();
+    };
+
+    const buttons = new Buttons(props, refresh);
+    updateToolbar({
+      actions: buttons
+        .expandSections(() => setCollapseSections({}))
+        .collapseSections(() =>
+          setCollapseSections({
+            [overviewSectionName]: true,
+            [paramsSectionName]: true,
+            [metricsSectionName]: true,
+          }),
+        )
+        .refresh(refresh)
+        .getToolbarActionMap(),
+      breadcrumbs: [{ displayName: 'Experiments', href: RoutePage.EXPERIMENTS }],
+      pageTitle: 'Compare runs',
+    });
+  }, [updateToolbar, refetch]);
+
+  useEffect(() => {
+    if (data) {
+      setSelectedIds(data.map(r => r.run!.id!));
+    }
+  }, [data]);
+
+  const collapseSectionsUpdate = (collapseSections: { [key: string]: boolean }): void => {
+    setCollapseSections({ ...collapseSections });
+  };
+
+  const showPageError = async (message: string, error: Error | undefined) => {
+    const errorMessage = await errorToMessage(error);
+    updateBanner({
+      additionalInfo: errorMessage ? errorMessage : undefined,
+      message: message + (errorMessage ? ' Click Details for more information.' : ''),
+    });
+  };
+
+  const selectionChanged = (selectedIds: string[]): void => {
+    setSelectedIds(selectedIds);
+  };
+
+  if (isError) {
+    return <></>;
+  }
 
-function CompareV2() {
   return (
-    <div className={commonCss.page}>
-      <p>This is the V2 Run Comparison page.</p>
+    <div className={classes(commonCss.page, padding(20, 'lrt'))}>
+      {/* Overview section */}
+      <CollapseButton
+        sectionName={overviewSectionName}
+        collapseSections={collapseSections}
+        collapseSectionsUpdate={collapseSectionsUpdate}
+      />
+      {!collapseSections[overviewSectionName] && (
+        <div className={commonCss.noShrink}>
+          <RunList
+            onError={showPageError}
+            {...props}
+            selectedIds={selectedIds}
+            ref={runlistRef}
+            runIdListMask={runIds}
+            disablePaging={true}
+            onSelectionChange={selectionChanged}
+          />
+        </div>
+      )}
+
+      <Separator orientation='vertical' />
+
+      {/* Parameters section */}
+      <CollapseButton
+        sectionName={paramsSectionName}
+        collapseSections={collapseSections}
+        collapseSectionsUpdate={collapseSectionsUpdate}
+      />
+      {!collapseSections[paramsSectionName] && (
+        <div className={classes(commonCss.noShrink, css.outputsRow)}>
+          <Separator orientation='vertical' />
+          <p>Parameter Section V2</p>
```

Comment:
Yes, that's only for temporary use, I'll remove this once I implement the Parameter section (mentioned in the Notes section above):

> The tests in `Compare.test.tsx` are updated because the `Run overview` text now exists on both the V1 and V2 pages. Since I won't implement the Parameter section until the second half of the internship, I used the existence or absence of that temporary text (`Parameter Section V2`) to test whether the V1 or V2 page is shown (I will change this again after I add the Parameter section).

Reply:
Sounds good! Resolving this now